src/test/libcephfs/test.cc

```diff
@@ -3017,6 +3017,17 @@ TEST(LibCephFS, Readlinkat) {
   target[target_len] = '\0';
   ASSERT_EQ(0, memcmp(target, rel_file_path, target_len));
 
+#if defined(__linux__) && defined(O_PATH)
+  // test readlinkat with empty pathname relative to O_PATH|O_NOFOLLOW fd
```

Comment:
old "fd" need to be closed IMO.

Reply:
Rearranged and added a close for the fd being opened with `O_PATH | O_NOFOLLOW`.